opentelemetry-sdk/src/opentelemetry/sdk/_metrics/measurement_consumer.py

```diff
@@ -0,0 +1,29 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from abc import ABC, abstractmethod
+
+from opentelemetry.sdk._metrics.measurement import Measurement
+
+
+class MeasurementConsumer(ABC):
+    @abstractmethod
+    def consume(self, measurement: Measurement):
```

Comment:
Can you add all the methods that were in the design?

```py
class MeasurementConsumer(ABC):
    @abstractmethod
    def consume_measurement(self, measurement: Measurement) -> None:
        pass

    @abstractmethod
    def register_async_instrument(
        self, instrument: SDKAsyncInstrumentBase
    ) -> None:
        pass

    @abstractmethod
    def collect(
        self, metric_reader: MetricReader, temporality: AggregationTemporality
    ) -> Iterable[Metric]:
        pass
```

you can wire them into the MeterProvider and test that:
- consume_measurement on the mock gets called whenever a sync instruement records a measurement
- register_async_instrument gets called whenver an async instrument is created

Reply:
Done